### PR TITLE
Improve `ForkCacheContent` types

### LINKED ISSUE #1848: Improve `ForkCacheContent` types
### Which components does the task require to be changed? (think hard pls)

snforge

### Description

Currently `ForkCacheContent` uses hashmaps of strings eg. `class_hash_at: HashMap<String, String>`, use appropriate types instead

### PR DESCRIPTION
**Pull request by Draggu**
Closes #1848

## Introduced changes

- Improves `ForkCacheContent` types
- ^ changes way it is serialized

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by Arcticae — APPROVED** [review]

File: crates/cheatnet/src/forking/cache.rs [hunk 1]
```diff
@@ -94,28 +99,24 @@ impl ForkCache {
...
-                ForkCacheContent::from_str(cache_file_content.as_str())
-            };
+        let cache_file = cache_file_path_from_fork_config(url, block_number, cache_dir)?;
+        let mut file = OpenOptions::new()
+            .write(true)
+            .read(true)
+            .create(true)
+            .truncate(false)
+            .open(&cache_file)
+            .unwrap();
```
Comment (piotmag769) [inline crates/cheatnet/src/forking/cache.rs:109]: Can we bubble an error here?

File: crates/cheatnet/src/forking/cache.rs [hunk 2]
```diff
@@ -94,28 +99,24 @@ impl ForkCache {
...
+
+        let mut cache_file_content = String::new();
+        file.read_to_string(&mut cache_file_content)
+            .expect("Could not read cache file: {path}");
```
Comment (piotmag769) [inline crates/cheatnet/src/forking/cache.rs:113]: Vide supra

File: crates/cheatnet/src/forking/cache.rs [hunk 3]
```diff
@@ -125,26 +126,22 @@ impl ForkCache {
...
-            .cache_file
-            .as_ref()
-            .unwrap_or_else(|| panic!("No cache_file to save to"));
         let mut file = OpenOptions::new()
             .write(true)
             .create(true)
             .truncate(false)
-            .open(cache_file)
+            .open(&self.cache_file)
             .unwrap();
```
Comment (piotmag769) [inline crates/cheatnet/src/forking/cache.rs:134]: vide supra

**Review by piotmag769 — APPROVED** [review]
We should handle the OS errors gracefully if possible

File: crates/cheatnet/src/forking/cache.rs (hunk 3 above)
Comment (Draggu) [reply crates/cheatnet/src/forking/cache.rs:134]: https://github.com/foundry-rs/starknet-foundry/pull/1940#issuecomment-2022471717

File: crates/cheatnet/src/forking/cache.rs (hunk 1 above)
Comment (Draggu) [reply crates/cheatnet/src/forking/cache.rs]: Fixed

File: crates/cheatnet/src/forking/cache.rs (hunk 2 above)
Comment (Draggu) [reply crates/cheatnet/src/forking/cache.rs]: Fixed

**Pull request merged**